R/Data-validity.R

```diff
@@ -16,15 +16,18 @@ NULL
 #'   `ID` and `cohort` vectors are of the same length equal to `nObs`.
 #'
 validate_subjects <- function(object) {
-  assert_class(object, "GeneralData")
-  assert_integer(object@nObs, len = 1L)
-  assert_integer(object@ID)
-  assert_integer(object@cohort)
-  
   o <- Validate()
+  
+  o$check(is.integer(object@nObs), "nObs must be of type integer")
+  o$check(length(object@nObs) == 1L, "nObs must be of length 1")
```

Comment:
actually this you can simplify using `test_int(object@nObs)` - so there are specific test functions for scalars of certain types in checkmate